qiskit_machine_learning/primitives/kernels/base_kernel.py

```diff
@@ -0,0 +1,117 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base kernel"""
+
+from abc import abstractmethod
+from typing import Tuple
+
+import numpy as np
+
+
+class BaseKernel:
+    """
+    Abstract class providing the interface for the quantum kernel classes.
+    """
+
+    def __init__(self, *, enforce_psd: bool = True) -> None:
+        """
+        Args:
+            sampler_factory: A callable that creates a qiskit primitives sampler
+                given a list of circuits.
+            enforce_psd: Project to closest positive semidefinite matrix if x = y.
+                Default True.
+        """
+        self._num_features: int = 0
+        self._enforce_psd = enforce_psd
+
+    @abstractmethod
+    def evaluate(self, x_vec: np.ndarray, y_vec: np.ndarray = None) -> np.ndarray:
+        r"""
+        Construct kernel matrix for given data
+
+        If y_vec is None, self inner product is calculated.
+
+        Args:
+            x_vec: 1D or 2D array of datapoints, NxD, where N is the number of datapoints,
+                                                            D is the feature dimension
+            y_vec: 1D or 2D array of datapoints, MxD, where M is the number of datapoints,
+                                                            D is the feature dimension
+
+        Returns:
+            2D matrix, NxM
+
+        Raises:
+            QiskitMachineLearningError:
+                A quantum instance or backend has not been provided
```

Comment:
```suggestion
                A quantum instance or backend has not been provided.
```
Please check other places.